src/model_ode.cpp

```diff
@@ -337,13 +390,41 @@ void Model_ODE::fsxdot(const realtype t, const AmiVector &x,
 
 void Model_ODE::fsxdot(realtype t, N_Vector x, int ip, N_Vector sx,
                        N_Vector sxdot) {
+    
+    /* sxdot is just the total derivative d(xdot)dp,
+       so we just call dxdotdp and copy the stuff over */
     if (ip == 0) {
         // we only need to call this for the first parameter index will be
         // the same for all remaining
         fdxdotdp(t, x);
         fJSparse(t, x, J.get());
     }
-    N_VScale(1.0, dxdotdp.getNVector(ip), sxdot);
+    if (pythonGenerated) {
+        /* copy dxdotdp and the implicit version over */
+        // initialize
+        N_VConst(0.0, sxdot);
+        realtype *sxdot_tmp = N_VGetArrayPointer(sxdot);
+        
+        // copy explicit version
+        if (ndxdotdp_explicit > 0) {
+            auto col_exp = dxdotdp_explicit.indexptrs();
+            for (sunindextype i = col_exp[ip]; i <  col_exp[ip + 1]; ++i)
+                sxdot_tmp[i] += (dxdotdp_explicit.data())[i];
```

Comment:
pretty sure there needs to be some indexing with `indexvals()` here aswell.

Reply:
It does indeed, yep. It just survived the testing as `dxdotdp_explicit` is in very most cases empty...